swarm/network/stream/snapshot_sync_test.go

```diff
@@ -0,0 +1,568 @@
+	defer localStoreCleanup()
+
+	trigger := make(chan discover.NodeID)
+	//triggerCheck defines what will be checked during the test
+	triggerCheck := func(ctx context.Context, id discover.NodeID) (bool, error) {
+		select {
+		case <-ctx.Done():
+			return false, ctx.Err()
+			//case <-disconnectC:
+			//  log.Error("Disconnect event detected")
+			//  return false, ctx.Err()
+		default:
+		}
+
+		log.Debug(fmt.Sprintf("Checking node: %s", id))
+		//select the local store for the given node
+		lstore := stores[id]
+		//if there are more than one chunk, test only succeeds if all expected chunks are found
+		allSuccess := true
+		//this selects which chunks are expected to be found with the given node
+		//localChunks := nodesToChunksMap[id]
+		localChunks := conf.nodesToChunksMap[string(conf.idToAddrMap[id])]
+
+		//for each expected chunk, check if it is in the local store
+		for i := 0; i < len(localChunks); i++ {
+			//ignore zero chunks
+			chunk := conf.chunks[localChunks[i]]
+			if storage.IsZeroKey(chunk) {
+				continue
+			}
+			log.Debug(fmt.Sprintf("node has chunk: %s:", chunk))
+			if _, err := lstore.Get(chunk); err != nil {
+				log.Error(fmt.Sprintf("Chunk %s NOT found for id %s", chunk, id))
+				allSuccess = false
+			} else {
+				fmt.Println("^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^")
+				log.Info(fmt.Sprintf("Chunk %s FOUND for id %s", chunk, id))
+			}
+		}
+
+		return allSuccess, nil
+	}
+
+	timeout := 300 * time.Second
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	//define the action to be performed before the test checks: start syncing
+	action := func(ctx context.Context) error {
+		// need to wait till an aynchronous process registers the peers in streamer.peers
+		// that is used by Subscribe
+		// the global peerCount function tells how many connections each node has
+		// TODO: this is to be reimplemented with peerEvent watcher without global var
+		i := 0
+		for err := range waitPeerErrC {
+			if err != nil {
+				return fmt.Errorf("error waiting for peers: %s", err)
+			}
+			i++
+			if i == len(ids)-1 {
+				break
+			}
+		}
+
+		time.Sleep(10 * time.Second)
+		// each node Subscribes to each other's swarmChunkServerStreamName
+		for j := 0; j < len(ids); j++ {
+			log.Debug(fmt.Sprintf("subscribe: %d", j))
+			ctx, cancel := context.WithTimeout(ctx, 1*time.Second)
+			defer cancel()
+			client, err := net.GetNode(ids[j]).Client()
+			if err != nil {
+				return err
+			}
+			err = client.CallContext(ctx, nil, "stream_startSyncing")
+			if err != nil {
+				log.Error(fmt.Sprintf("FAILED CallContext %v", err))
+				return nil
+			}
+		}
+		time.Sleep(10 * time.Second)
+		//now upload the chunks to the selected random single node
+		conf.chunks, err = uploadFileToSingleNodeStore(node.ID(), chunkCount)
+		if err != nil {
+			return err
+		}
+		//finally map chunks to the closest addresses
+		conf = mapKeysToNodes(conf)
```

Comment:
no elegant to reassign conf here. just return the changed values and assign it directly to conf fields